### PR TITLE
[EVNT-601] Refactor camel integration routes

### PR DESCRIPTION
**Pull request by jpramos123**
* introduces `integrations.component.name` property which configures component name
  * this affects message event selection on the ingress topic
* separates error handling route
  * Thanks to common Camel Context routing between route builder classes this works like a charm.
* separates common Route stuff to `IntegrationsRouteBuilder` base/abstract class
  * this takes care of exception registration
  * and common constants
* separate main routes from a specific handler ones to `MainRoutes` class
  * Splunk specific handler is separated
  * main routing including ingress and return channel stays on `MainRoutes`
  * route names are kept
  * this would welcome new integrations
* sets `camel.main.javaRoutesIncludePattern` which selects route builders
  * this would enable switching between different concrete integrations
  * https://camel.apache.org/components/3.17.x/others/main.html
* reogranizes package names to `com.redhat.console.integration`
  * concrete Splunk integration is in the splunk subpackage

EVNT-601

### CODE REVIEW

**Comment by jpramos123** [comment]
@vkrizan My strategy was to create a `GenericIntegration` abstract class so other integrations can extend this class for a basic configuration.

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/GenericIntegration.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        // Decode CloudEvent
+        .process(new CloudEventDecoder())
+        // We check that this is our type.
+        // Otherwise, we ignore the message there will be another component that takes
+        // care
+        .log("IM WORKING HERE")
+        .filter().simple("${header.ce-type} == '" + CE_TYPE + "'")
+        // Log the parsed cloudevent message.
+        .to(log("com.redhat.console.notifications.splunkintegration?level=DEBUG"))
+        .to(direct("handler"))
```
Comment (jpramos123) [inline splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/GenericIntegration.java:166]: @vkrizan I'm not able to find a way to redirect to the child `handler` route

Do you have any idea?

Thanks

**Comment by jpramos123** [comment]
@vkrizan I would like to go with the subclass approach... In this way the Integration classes can inherit from the `BaseRoutes` the error handlers (any new error handler will be simply added to the `BaseRoutes` class in future) and the Integration classes can keep their specific handler implementation. 

Other wise, every new Integration will need to have the its own `configureErrorHandler();` method, and any new error handler will be needed to be implemented in all Integrations.

Below is the approach:
```
BaseRoutes
  - configures every routes but the `handler` + error handlers
SplunkIntegration
  - configures just the `handler` and inherit error handlers from BaseRoutes
SnowIntegration
  - configures just the `handler` and inherit error handlers from BaseRoutes

```

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/ErrorHandler.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+import org.apache.camel.Exchange;
+import org.apache.camel.LoggingLevel;
+import org.apache.camel.Processor;
+import org.apache.camel.builder.endpoint.EndpointRouteBuilder;
+import org.apache.camel.http.base.HttpOperationFailedException;
+import org.apache.http.ProtocolException;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+abstract class ErrorHandler extends EndpointRouteBuilder {
```
Comment (vkrizan) [inline splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/ErrorHandler.java:13]: Maybe use name like `ErrorHandlingRouteBase`.

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/ErrorHandler.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+  
+
```
Comment (vkrizan) [inline splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/ErrorHandler.java:16]: There are too many empty lines. Try running a formatter `mvnw formatter:format`.

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/ErrorHandler.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+  public String LOGGER_NAME;
+  
+  public static final String RETURN_TYPE = "com.redhat.console.notifications.history";
+
+  @Override
+  public void configure() throws Exception {
+    LOGGER_NAME = "com.redhat.console.notification.toCamel." + COMPONENT_NAME;
+
+    getContext().getGlobalOptions().put(Exchange.LOG_EIP_NAME, LOGGER_NAME);
```
Comment (vkrizan) [inline splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/ErrorHandler.java:28]: This global setting is redefined second time. I think it would affect it globally within the Camel Context. We can have it defined once on BaseRoutes.

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+
+1) Adapt code to receive parameter from env variables
+2) Add parameters to application.properties
+3) Refact errorHandlers
+4) Refact ingress
+5) Refact return
+6) Refact success
+
+
+*/
```
Comment (vkrizan) [inline splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java:19]: Please drop the notes from comments.

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+  String kafkaReturnGroupId;
+  
+  // The return type
+  public static final String RETURN_TYPE = "com.redhat.console.notifications.history";
+
+  @Override
+  public void configure() throws Exception {
+    super.configure();
+
+    CE_TYPE = "com.redhat.console.integration." + COMPONENT_NAME;
```
Comment (vkrizan) [inline splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java:59]: Incorrect type. It should be `"com.redhat.console.notification.toCamel." + COMPONENT_NAME`.
The filer is not working correctly otherwise.

Also can this be defined as a final constant?
There was `CONFIG` defined:
```
private static final Config CONFIG = ConfigProvider.getConfig();
```
that can getValue of a parameter.

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    LOGGER_NAME = "com.redhat.console.notification.toCamel." + COMPONENT_NAME;
```
Comment (vkrizan) [inline splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java:60]: The same as above. Can it be a final constant?

**Review by vkrizan — COMMENTED** [review]
Nice one!

TL;DR: formatting, filter, comments, constants

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/ErrorHandler.java (hunk 2 above)
Comment (jpramos123) [reply splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/ErrorHandler.java:13]: ACK! Moving with `ErrorHandlingBaseRoutes` I think this name makes more sense!

Thanks

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/ErrorHandler.java (hunk 4 above)
Comment (jpramos123) [reply splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/ErrorHandler.java:28]: Done! Keeping only in BaseRoutes class

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java (hunk 5 above)
Comment (jpramos123) [reply splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java:19]: Done!

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java (hunk 6 above)
Comment (jpramos123) [reply splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java:59]: Nice! I remember I saw this `CONFIG` parameter there, but it was not being used!

Now I could apply it and define other fields as `static final`

Thanks!

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java (hunk 7 above)
Comment (jpramos123) [reply splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java:60]: Done!

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+@ApplicationScoped
+public class BaseRoutes extends ErrorHandlingBaseRoutes {
```
Comment (vkrizan) [inline splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/BaseRoutes.java:12]: This should extend from `EndpointRouteBuilder` as the `ErrorHandlingBaseRoutes` are not defined yet.

File: splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java [hunk 9]
```diff
@@ -96,16 +70,12 @@ protected void initialize() {
...
-
         configureErrorHandler();
-        configureIngress();
         configureIoFailed();
         configureHttpFailed();
         configureTargetUrlValidationFailed();
         configureSecureConnectionFailed();
-        configureReturn();
-        configureSuccessHandler();
+        super.configure();
```
Comment (vkrizan) [inline splunk-quarkus/src/main/java/com/redhat/console/notifications/splunkintegration/SplunkIntegration.java:73]: `super` not required at this stage yet.

**Comment by vkrizan** [comment]
@petrblaho can you please have a look?

**Comment by petrblaho** [comment]
The code looks ok, I did not find anything wrong on the first look.

**Review by petrblaho — APPROVED** [review]
LGTM!
